src/corehost/cli/test/nativehost/resolve_component_dependencies_test.cpp

```diff
@@ -0,0 +1,208 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "resolve_component_dependencies_test.h"
+#include "hostfxr_exports.h"
+#include <error_codes.h>
+#include "hostpolicy_exports.h"
+#include "error_writer_redirector.h"
+#include <future>
+
+namespace
+{
+    template<int n>
```

Comment:
Is this a template also for the different threads? Would `thread_local` work? The templates as a means to explicitly/manually differentiate threads is non-obvious to me.

Reply:
Same as above - changed to thread local.